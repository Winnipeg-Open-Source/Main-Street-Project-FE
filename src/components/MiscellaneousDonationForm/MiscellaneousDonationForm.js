import React, { memo } from 'react';
import PropTypes from 'prop-types';
import Input from 'components/Input';
import TextArea from 'components/TextArea';

function MiscellaneousDonationForm ({
    name,
    description,
    quantity,
    notes,
    onChange,
}) {
    return (
        <>
            <Input id='name' name='name' placeholder='Name' value={name} onChange={onChange} />
            <Input id='description' name='description' placeholder='Description' value={description} onChange={onChange} />
            <Input id='quantity' name='quantity' placeholder='Quantity' value={quantity} onChange={onChange} />
            <TextArea id='notes' name='notes' label='Notes' value={notes} onChange={onChange} />
        </>
    );
}

MiscellaneousDonationForm.displayName = 'MiscellaneousDonationForm';

MiscellaneousDonationForm.propTypes = {
    name: PropTypes.string,
    description: PropTypes.string,
    quantity: PropTypes.number,
    notes: PropTypes.string,
    onChange: PropTypes.func,
};

export default memo(MiscellaneousDonationForm);
