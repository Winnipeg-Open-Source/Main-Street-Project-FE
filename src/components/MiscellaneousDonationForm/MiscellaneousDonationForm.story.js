import React from 'react';
import { action } from '@storybook/addon-actions';
import Form from 'components/Form';
import MiscellaneousDonationForm from './MiscellaneousDonationForm';
import { mockLocations } from 'tests/mocks/locations';
import { mockLineItems } from 'tests/mocks/items';

const props = {
    name: 'Misc Name',
    description: 'A miscellaneous item',
    quantityName: 'Rolls',
    locations: mockLocations,
    lineItems: mockLineItems,
    notes: 'Something I had lying around that I do not need anymore',
    onChange: action('on change'),
    onAddLineItemClick: action('on line item clicked'),
    onItemQuantityChange: action('on line item quantity changed'),
};

export const EmptyForm = () => (
    <Form>
        <MiscellaneousDonationForm />
    </Form>
);

export const CompletedForm = () => (
    <Form>
        <MiscellaneousDonationForm {...props} />
    </Form>
);

export default {
    title: 'Forms|MiscellaneousDonationForm',
    component: MiscellaneousDonationForm,
};
