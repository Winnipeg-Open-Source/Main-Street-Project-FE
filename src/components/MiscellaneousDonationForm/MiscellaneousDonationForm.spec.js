import React from 'react';
import { render } from '@testing-library/react';
import MiscellaneousDonationForm from './MiscellaneousDonationForm';

const props = {
    name: 'Misc Name',
    description: 'A miscellaneous item',
    quantity: 3,
    notes: 'Something I had lying around that I do not need anymore',
    onNameChange: () => {},
    onDescriptionChange: () => {},
    onQuantityChange: () => {},
    onNotesChange: () => {},
};

describe ('MiscellaneousDonationForm', () => {
    it ('renders without crashing', () => {
        const { asFragment } = render(<MiscellaneousDonationForm />);

        expect(asFragment()).toMatchSnapshot();
    });

    it ('has placeholders when empty', async () => {
        const { getByLabelText, getByPlaceholderText } = render(<MiscellaneousDonationForm />);

        const name = await getByPlaceholderText('Name');
        const description = await getByPlaceholderText('Description');
        const quantity = await getByPlaceholderText('Quantity');
        const notes = await getByLabelText('Notes');

        expect(name.value).toBe('');
        expect(description.value).toBe('');
        expect(quantity.value).toBe('');
        expect(notes.value).toBe('');
    });

    it ('has all fields used', async () => {
        const { getByLabelText, getByPlaceholderText } = render(<MiscellaneousDonationForm {...props} />);

        const name = await getByPlaceholderText('Name');
        const description = await getByPlaceholderText('Description');
        const quantity = await getByPlaceholderText('Quantity');
        const notes = await getByLabelText('Notes');

        expect(name.value).toBe('Misc Name');
        expect(description.value).toBe('A miscellaneous item');
        expect(quantity.value).toBe('3');
        expect(notes.value).toBe('Something I had lying around that I do not need anymore');
    });
});
