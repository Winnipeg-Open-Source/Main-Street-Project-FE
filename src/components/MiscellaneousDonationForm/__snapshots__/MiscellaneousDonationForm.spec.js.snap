// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiscellaneousDonationForm renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="sc-qOvoE hIfWnJ sc-qWSGh dpgLb"
  >
    <input
      class="sc-pdkfH sc-pAMRk fgScIC"
      id="name"
      name="name"
      placeholder="Name"
      value=""
    />
    <input
      class="sc-pdkfH sc-pAMRk fgScIC"
      id="description"
      name="description"
      placeholder="Description"
      value=""
    />
    <input
      class="sc-pdkfH sc-pAMRk fgScIC"
      id="quantity-name"
      name="quantityName"
      placeholder="Quantity Name"
      value=""
    />
    <div
      class="sc-AxirZ sc-AxhCb sc-prPLn hvsTFl"
      width="1"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="downshift-0-label"
        role="combobox"
        style="position: relative; width: 100%;"
      >
        <input
          aria-autocomplete="list"
          aria-labelledby="downshift-0-label"
          autocomplete="off"
          class="sc-pdkfH jYnISK sc-pZCbX jZUHIk"
          font-size="2,,1"
          id="downshift-0-input"
          placeholder="Location Search"
          value=""
          width="1"
        />
        <div
          aria-labelledby="downshift-0-label"
          class="sc-AxirZ sc-pQfvp sc-pIITJ hkyCZe"
          height="256px"
          id="downshift-0-menu"
          role="listbox"
        />
      </div>
    </div>
    <label
      class="sc-ptfmh cDYtMS"
      color="border.light"
      font-size="10px"
      font-weight="bold"
    >
      Notes
      <textarea
        class="sc-pTUxa sc-qYhBd fKpFqM"
        id="notes"
        name="notes"
      />
    </label>
  </div>
</DocumentFragment>
`;
